python-test-samples/lambda-mock/README.md

```diff
@@ -0,0 +1,141 @@
+
+This pattern can be used with a variety of infrastructure as code systems including SAM, Serverless Framework, CDK, CloudFormation and Terraform. This pattern uses a simple test framework, with the test harness directly calling the Lambda function handlers.  No cloud resources or full ** stack emulation are required.
+
+![Test Pattern](/_img/pattern_04_lambda_mock_test.png)
+
+[Top](#contents)
+
+---
+
+## Sample Project
+
+For this example, we demonstrate a system that generates a document, written to Amazon S3, based on contents from a key-value lookup in DynamoDB.
+
+![Architecture](doc/architecture.png)
+
+An API Gatway path [1] triggers an AWS Lambda function [2] that retrieves a data from a DynamoDB [3] table and writes data to a object on Amazon S3 [4].  The API path contains a Document Type and a Customer ID.  The Lambda function retrieves both the Document Type data and Customer ID data and combines them, writing the data to S3 and returning the object key [1].  The DynamoDB table name and the S3 bucket name are provided to the Lambda function via environment variables.  
+
+The DynamoDB table schema is comprised of a Partition Key (PK) for looking up a given item, and a “data” field containing string contents.  Document Type Items are prefixed with D#, and Customer items have a PK prefixed with C#.
+
+[Top](#contents)
+
+---
+
+## Key Files in the Project
+  - [app.py](src/sample_lambda/app.py) - Lambda handler code to test
+  - [test_sample_lambda.py](tests/unit/src/test_sample_lambda.py) - Unit test using mocks
+  - [template.yaml](template.yaml) - SAM script for deployment
+  - [Makefile](Makefile) - Commands for setup, test, and deployment
+  
+
+> The code in this project is simplified - we’ve opted for concise snippets over fully completed or PEP8 compliant production code.
+
+[Top](#contents)
+
+---
+## Prerequisites
+
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+* [AWS Serverless Application Model (SAM) CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* [Python 3.9](https://www.python.org/downloads/)
+* [Make build utility](https://www.gnu.org/software/make/)
+
+[Top](#contents)
+
+---
+
+## Make commands for test and deployment
+
+The project Makefile contains helper commands for working with the project:
+* ```make install```: Create a Python Virtual Envionment and install dependencies
+* ```make test```: Run a unit test, guarding for socket connections
+* ```make coverage```: Run unit tests and provide a coverage report
+* ```make deploy```: Deploy the stack to an AWS Account
+* ```make deploy.g```: Deploy the stack to an AWS Account, prompting for stack parameters
+
+[Top](#contents)
+
+---
+
+## Running the project
+
+* First, create the Python Virtual Environment by using the ```make createEnv``` command.
```

Comment:
wrong make command?

Reply:
I missed a whole section of the README when I re-tooled the Makefile, thank you for finding it.   Resolved.